lib/debug/thread_client.rb

```diff
@@ -576,13 +563,31 @@ def wait_next_action
           end
 
           event! :result, nil
-
         when :breakpoint
-          add_breakpoint args
+          case args[0]
+          when :method
+            bp = initialize_breakpoint args
+            event! :result, :method_breakpoint, bp
+          when :watch
+            ivar = args[1]
+            result = frame_eval(ivar)
 
+            if @success_last_eval
+              object =
+                if b = current_frame.binding
+                  b.receiver
+                else
+                  current_frame.self
+                end
+              puts "#{object} #{ivar} = #{result}"
```

Comment:
I think this can go now because it's the same content as the breakpoint's string representation.

```
#<Student:0x00007fa8f99465c0> @name = John # printed by this line 
#0  BP - Watch  #<Student:0x00007fa8f99465c0> @name = John # printed by show_bps
```

Reply:
@ko1 I've added the last commit to remove it.